### PR TITLE
Assigment4,work in progress

### PR DESCRIPTION
**Pull request by Denrig**

### CODE REVIEW

File: app/assets/stylesheets/shared/variables.scss [hunk 1]
```diff
@@ -0,0 +1,30 @@
+//SORTING BAR
+$sorting-background-color: #a0999a88;
+
+$sorting-font: Verdana;
+$sorting-font-color: rgb(0, 0, 0);
+
+//PRODUCTS DISPLAY
+$name-background-color: hsla(0, 0%, 35%, 0.137);
```
Comment (RobertCristian) [inline app/assets/stylesheets/shared/variables.scss:8]: I would convert this to rgba just to maintain a level of consistency.

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    @user = User.new
+  end
+
+  def show
+    @user = User.find(params[:id])
+  end
+
+  def create
+    @user = User.new(user_params)
+    if @user != User.new && @user.save
```
Comment (RobertCristian) [inline app/controllers/users_controller.rb:12]: Why the `@user != User.new` check ?

File: app/assets/stylesheets/sessions.scss [hunk 3]
```diff
@@ -0,0 +1,21 @@
+// Place all the styles related to the Sessions controller here.
+// They will automatically be included in application.css.
+// You can use Sass (SCSS) here: https://sass-lang.com/
+
+@import "./shared/variables.scss";
```
Comment (RobertCristian) [inline app/assets/stylesheets/sessions.scss:5]: How would you do this so you don't load `variables.scss` in (almost) every .scss files?

File: app/controllers/sessions_controller.rb [hunk 4]
```diff
@@ -0,0 +1,20 @@
+class SessionsController < ApplicationController
+  def new; end
+
+  def create
+    user = User.find_by(email: params[:session][:email].downcase)
+    if user&.authenticate(params[:session][:password])
+      reset_session
```
Comment (RobertCristian) [inline app/controllers/sessions_controller.rb:7]: Why is a session reset needed at login?

File: app/javascript/packs/application.js [hunk 5]
```diff
@@ -4,6 +4,8 @@
 // that code so it'll be compiled.
 
 require("@rails/ujs").start()
+require("turbolinks").start()
 require("@rails/activestorage").start()
 require("channels")
-require('jquery')
\ No newline at end of file
+require("jquery")
+import 'bootstrap'
```
Comment (RobertCristian) [inline app/javascript/packs/application.js:11]: don't forget to always add a new line at the end.

File: app/controllers/sessions_controller.rb (hunk 4 above)
Comment (Denrig) [reply app/controllers/sessions_controller.rb:7]: It prevents session hijacking,by making sure that there is no session active with our current user id

**Pull request closed without merging**